src/components/dashboard/DashboardPage.jsx

```diff
@@ -57,6 +58,7 @@ const DashboardPage = () => {
     <Container size="lg" className="py-4">
       <Helmet title={PAGE_TITLE} />
       <BudgetExpiryNotification />
+      <ExpiredSubscriptionModal />
```

Comment:
[suggestion] Now that there are multiple components in this file related to expired subscriptions, e.g. see the following below around line ~100:

```jsx
{subscriptions.subscriptionPlan && subscriptions.showExpirationNotifications && <SubscriptionExpirationModal />}
```

... it might be a bit confusing to have 2 different components with similar responsibilities (i.e., `ExpiredSubscriptionModal` and `SubscriptionExpirationModal`), especially when rendered in different places within `DashboardPage`.

A couple considerations:
1. If we do keep both as separate modals, I might recommend rendering the related modal components together, co-located, within `DashboardPage`, ideally with explicit/clear documentation (e.g., comments, perhaps an ADR) *why* there's multiple related components pertaining to subscription expiration.
2. Consider extending/re-purposing the existing `SubscriptionExpirationModal` to include the custom modal messaging use case. The component is currently responsible for showing the default "expiring" or "expired" subscriptions modal, rendered only on the Dashboard page today. Perhaps it could also be responsible for showing the custom subscription expiration messaging, too?
    * For example, for a learner with an expired subscription license, when custom subscription expiration messaging is configured, show the custom expiration modal or otherwise fallback to the default expiration modal if no custom messaging is configured. That said, if we go this route, I might recommend creating sub-components within `SubscriptionExpirationModal` to avoid making the single component that exists today larger / difficult to parse (e.g., `CustomSubscriptionExpiredModal`, `SubscriptionExpiredModal`, `SubscriptionExpiringModal`).
    * However, while easier to reason about, this approach would likely mean showing the existing subscription expiration messaging on the search/course routes where it is not present today. That said, adding the existing expiration modal messaging to these additional routes is likely actually desirable.
    * Implementation-wise, the conditional rendering of `SubscriptionExpirationModal` that exists today (i.e., `subscriptions.subscriptionPlan && subscriptions.showExpirationNotifications`) would also need to move inside `SubscriptionExpirationModal` in order to support the custom expiration modal use case.

Reply:
@adamstankiewicz I agree with your suggestion!
We can create a follow-up ticket to repurpose the existing `SubscriptionExpirationModal` to handle the custom subscription expiration messaging. For now, we can render the related modal components together, with clear comments explaining the purpose.

Reply:
@adamstankiewicz A [ticket](https://2u-internal.atlassian.net/browse/ENT-9650) has been created for this purpose.

Reply:
Thanks for the ticket link!